### DISCUSSION TITLE
Date Times

### DISCUSSION THREAD

**Discussion by ByronMayne** [LEGO/AsyncAPI.NET#151] [General]
Hey @VisualBean,

I was working on fixing up the unit tests and trying to solve some of the build issues. A lot of the complexity comes from converting date times. For example the unit tests fail on linux due to the `GetYamlCompatibleString` function. In side this code we have a block that attempts to convert to date time. 

```cs
if (decimal.TryParse(input, NumberStyles.Float, CultureInfo.InvariantCulture, out var _) ||
    bool.TryParse(input, out var _) ||
    DateTime.TryParse(input, out var _))
{
    return $"'{input}'";
}
```
The problem here is that depending on the platform and the numbers used this produces different results 

| Input | Platform | TryParse |  Serialized As |
|-------|-----------|--------|------------------|
| `"3.2.1"` | Win | `true` | `'3.2.1'` |
| `"3.2.1"` | Linux | `true` | `'3.2.1'`|
| `"2.6.0"` | Win | `false` | `2.6.0` 💀 | 
| `"2.6.0"` | Linux | `true` | `'2.6.0'` |

That being the case, do I am thinking it's a good idea to be more strict about the formats of the dates. Then we can handle explicit cases rather then depending on the platform implementation.  

# Question 
Since both Javascript (for Json) and OpenApi say to use `ISO8601` why not unify behind that and remove this ambiguity that happens across platforms?

**Comment by VisualBean** [comment]
Thanks a ton, for spending the time investigating this.

I think it would be great to lean against iso and enforce this for datetime parsing.
Although not explicitly set for asyncapi, I think it would work much better for this lib and where we are going with it. 

Awesome work!

**Reply by ByronMayne** [reply]
No problem!

I am planning to use the library in my day job so I don't mind spending some time try to polish it up. 


I will start making the changes for ISO and update the unit tests to make sure it's explicitly guarded against.
